scipy/signal/_signaltools.py

```diff
@@ -3500,210 +3506,230 @@ def invresz(r, p, k, tol=1e-3, rtype='avg'):
 
-    Due to ``N`` being prime,
+    The following example illustrates that a `~scipy.fft.rfft` / `~scipy.fft.irfft`
+    combination does not always produce the correct resampling result, while
+    `resample` does:
 
-    >>> num = 5000
-    >>> f = signal.resample(signal.resample_poly(y, 1, N // num), num)
-
-    runs significantly faster than
+    >>> import numpy as np
+    >>> from scipy.fft import irfft, fft, fftfreq, rfft
+    >>> from scipy.signal import resample
+    ...
+    >>> n0, n1 = 8, 4  # number of samples
+    >>> t0, t1 = (np.linspace(0, 2, n_, endpoint=False) for n_ in (n0, n1))  # in s
+    >>> x0, x1 = (np.cos(2*np.pi*t_) for t_ in (t0, t1))  # 1 Hz cosine signal
+    >>> y1_r = resample(x0, num=n1)
+    >>> np.allclose(y1_r, x1)  # correct result, as expected
+    True
+    >>> y1_f = irfft(rfft(x0), n=n1) * n1/n0
+    >>> np.allclose(y1_f, x1)  # wrong result
+    False
+    >>> # Compare FFTs:
+    >>> fftfreq(n0, t0[1]-t0[0])  #  frequencies of fft(x0)
+    array([ 0. ,  0.5,  1. ,  1.5, -2. , -1.5, -1. , -0.5])
+    >>> np.round(fft(x0), 3)  # FFT of x0
+    array([-0.-0.j, -0.+0.j,  4.-0.j,  0.+0.j,  0.-0.j,  0.-0.j,  4.+0.j, -0.-0.j])
+    >>> fftfreq(n1, t1[1]-t1[0])  # frequencies of fft(x1) and fft(y1_f)
+    array([ 0. ,  0.5, -1. , -0.5])
+    >>> np.round(fft(x1), 3)  # reference FFT
+    array([0.+0.j, 0.+0.j, 4.+0.j, 0.+0.j])
+    >>> np.round(fft(y1_f), 3)  # irfft/rfft off by factor 2 in the Nyuist frequency
+    array([0.+0.j, 0.+0.j, 2.+0.j, 0.+0.j])
+
+    The reason for the different results lies in `resample` correctly treating unpaired
+    frequency bins. I.e., the input `x1` has a bin pair ±1 Hz, whereas the output has
+    only one unpaired bin at -1 Hz, which demands rescaling of that bin. Special
+    treatment is required if ``n_x != num`` and ``min(n_x, num)`` is even. If the bin
+    values at `±m` are zero, the, obviously, no special treatment is needed. Consult
+    the source code of `resample` for details.
+
+    The following code snippet shows how to use `resample_poly` to speed up the
+    down-sampling:
 
-    >>> f = signal.resample(y, num)
+    >>> import numpy as np
+    >>> from scipy.signal import resample, resample_poly
+    ...
+    >>> n0 = 19937 # number of input samples - prime
+    >>> n1 = 128  # number of output samples - fast FFT length
+    >>> x0 = np.random.rand(n0)  # input signal
+    ...
+    >>> y1 = resample(x0, n1)  # slow due to n0 being prime
+    >>> # This is faster:
+    >>> y1_p = resample(resample_poly(x0, 1, n0 // n1, padtype='wrap'), n1)
+
+    Note that the `y1` and `y1_p` are not identical due to the differences in the
+    utilized antialiasing filter in each function. In both functions the antialiasing
+    filter can be customized by modifying the `window` parameter, though `resample`
+    interprets it as a spectral window function, whereas `resample_poly` assumes it to
+    be some form of impulse response.
```

Comment:
I would say something like this... the "some form of impulse response" is what caught my eye as being unclear. The purpose in `resample_poly` is to perform the antialiasing. If you pass a bad antialiasing filter, you risk aliasing there. There is no risk of aliasing in `resample` since the ideal brick-wall filter is always implicitly applied by the method itself.
```suggestion
    filter can be customized by modifying the `window` parameter. `resample`
    interprets it as a *spectral* window function that is always applied in addition to
    the implicit ideal (brick wall) antialiasing filter. `resample_poly` applies
    `window` as a FIR filter that gets applied in the *time* domain and should be
    designed appropriately to avoid aliasing.
```

Reply:
Addressed in commit [b3c1993](https://github.com/scipy/scipy/pull/22814/commits/b3c199349fd5cc424dbbaa77f3b615881156a666)